### PR TITLE
PWGHF: Add c*t for V0s in the LcToPK0s datamodel, tree creator and task

### PR DESCRIPTION
**Pull request by DanielSamitz**

### CODE REVIEW

**Comment by DanielSamitz** [comment]
@rutuparnarath can you please check that this is what you wanted.

File: PWGHF/DataModel/CandidateReconstructionTables.h [hunk 1]
```diff
@@ -533,6 +533,8 @@ DECLARE_SOA_DYNAMIC_COLUMN(PtV0Pos, ptV0Pos, //!
                            [](float px, float py) { return RecoDecay::pt(px, py); });
 DECLARE_SOA_DYNAMIC_COLUMN(PtV0Neg, ptV0Neg, //!
                            [](float px, float py) { return RecoDecay::pt(px, py); });
+DECLARE_SOA_DYNAMIC_COLUMN(V0Ct, v0ct, //!
```
Comment (vkucera) [inline PWGHF/DataModel/CandidateReconstructionTables.h:536]:
```suggestion
DECLARE_SOA_DYNAMIC_COLUMN(CtV0, ctV0, //!
```
This seems more consistent with the names of functions where you use this column.

**Review by vkucera — CHANGES_REQUESTED** [review]

File: PWGHF/DataModel/CandidateReconstructionTables.h (hunk 1 above)
Comment (vkucera) [reply PWGHF/DataModel/CandidateReconstructionTables.h:536]: Also please add documentation for the columns in that namespace

File: PWGHF/DataModel/CandidateReconstructionTables.h [hunk 2]
```diff
@@ -522,24 +522,26 @@ DECLARE_SOA_TABLE(HfCand2ProngMcGen, "AOD", "HFCAND2PMCGEN", //!
...
+DECLARE_SOA_EXPRESSION_COLUMN(Pz, pz, //! pz of candidate
                               float, 1.f * aod::hf_cand::pzProng0 + 1.f * aod::hf_cand::pzProng1);
 // DECLARE_SOA_DYNAMIC_COLUMN(M, m, [](float px0, float py0, float pz0, float px1, float py1, float pz1, const array<double, 2>& m) { return RecoDecay::M(array{array{px0, py0, pz0}, array{px1, py1, pz1}}, m); });
-DECLARE_SOA_DYNAMIC_COLUMN(PtV0Pos, ptV0Pos, //!
+DECLARE_SOA_DYNAMIC_COLUMN(PtV0Pos, ptV0Pos, //! pt of the positive V0 daughter
                            [](float px, float py) { return RecoDecay::pt(px, py); });
-DECLARE_SOA_DYNAMIC_COLUMN(PtV0Neg, ptV0Neg, //!
+DECLARE_SOA_DYNAMIC_COLUMN(PtV0Neg, ptV0Neg, //! pt of the negative V0 daughter
                            [](float px, float py) { return RecoDecay::pt(px, py); });
+DECLARE_SOA_DYNAMIC_COLUMN(CtV0, ctv0, //! c*t of the V0
```
Comment (vkucera) [inline PWGHF/DataModel/CandidateReconstructionTables.h:536]:
```suggestion
DECLARE_SOA_DYNAMIC_COLUMN(CtV0, ctV0, //! c*t of the V0
```
Almost :-)

**Review by vkucera — CHANGES_REQUESTED** [review]

**Comment by rutuparnarath** [comment]
> @rutuparnarath can you please check that this is what you wanted.

Hi @DanielSamitz . Thank you for including these variables in the treecreator. Yes, this is the one that I have requested.

File: PWGHF/DataModel/CandidateReconstructionTables.h [hunk 3]
```diff
@@ -553,6 +555,18 @@ auto invMassGammaToEE(const T& candidate)
   return candidate.m(array{RecoDecay::getMassPDG(kElectron), RecoDecay::getMassPDG(kElectron)});
 }
 
+template <typename T>
+auto ctV0K0Short(const T& candidate)
```
Comment (vkucera) [inline PWGHF/DataModel/CandidateReconstructionTables.h:559]:
```suggestion
auto ctV0K0s(const T& candidate)
```
probably better for consistency with `LcToK0sP`.

**Review by vkucera — CHANGES_REQUESTED** [review]

File: PWGHF/DataModel/CandidateReconstructionTables.h (hunk 3 above)
Comment (DanielSamitz) [reply PWGHF/DataModel/CandidateReconstructionTables.h:559]: The way it is now it is more consistent with https://github.com/AliceO2Group/O2Physics/blob/207d4626f94d0cedb0c00ec5d4f94a71d3feceaa/PWGHF/D2H/Tasks/taskLcToK0sP.cxx#L93 , then I should change also that. But this I have named like that to be consistent with the v0data table https://github.com/AliceO2Group/O2Physics/blob/207d4626f94d0cedb0c00ec5d4f94a71d3feceaa/PWGLF/DataModel/LFStrangenessTables.h#L126-L127
As you prefer, I don't have a strong opinion on that.

Comment (vkucera) [reply PWGHF/DataModel/CandidateReconstructionTables.h:559]: For direct references to LF tables, columns and functions, I am fine with keeping `K0Short`. This case is a new HF code function which has no link to any LF code.

**Pull request merged**